xtevent/_eventgenvars.ado

```diff
@@ -1,306 +1,500 @@
-	qui replace `kg' = `=`rwindow'+1' if __k >= `rwindow' & `touse'
-	qui levelsof `kg', loc(kgs)
-	
-	* If extrapolating a linear trend, exclude some of the event time dummies
-	loc komittrend ""
-	* di "`included'"
-	if "`trend'" != "" {
-		if `bin'!=1 {
-			di as err _n "Cannot extrapolate linear trend with non-binary policyvar."
-			exit 301
-		}
-	
-		* Generate the trend		
-		qui gen _ttrend = __k  if `touse'	
-		qui replace _ttrend = 0 if !inrange(_ttrend,`lwindow',`=`rwindow'') & `touse'
-		* qui replace _ttrend = 0 if _ttrend<`=`trend'' & `touse'
-		qui replace _ttrend = 0 if mi(_ttrend) & `touse'
-		la var _ttrend trend
-		
-		
-		* Exclude the coefficients that are restricted: Those on the negative values of the trend range, plus the endpoints
-		
-		loc komittrend ""	
-		
-		foreach klevel in `kgs' {
-			if (inrange(`klevel',`trend',-1) & `klevel' != -1) {
-				loc absk = abs(`klevel')
-				if `klevel'<0 loc plus = "m"
-				else loc plus "p"
-				loc l`plus'`absk' "`plus'`absk'"
-				loc toexc "_k_eq_`l`plus'`absk''"
-				loc included: list included - toexc
-				loc komittrend = "`komittrend' `klevel'"				
-			}
-		*noi di "***"
-		*noi di "`included'"
-		}
-		* loc toexc: word 1 of `included'
-		*loc included: list local included - toexc
-		*loc total: word count `included'
-		*loc toexc: word `total' of `included'
-		*loc included: list local included - toexc
-		* di "`included'"
-	}	
-	
-	* Save the names of included reg vars to extract these coefficients from the overall matrix later
-	
-	loc j=1
-	loc names ""
-	foreach var in `included' {
-		if `j'==1 loc names `""`var'""'
-		else loc names `"`names'.."`var'""'
-		loc ++ j
-	}	
-	return local included = "`included'"
-	return local names = `"`names'"'
-	return local komittrend= "`komittrend'"	
-end
-
-
-
-
-
-
-	
-	
\ No newline at end of file
+* _eventgenvars.ado 1.00 Aug 24 2021
+
+version 11.2
+
+cap program drop _eventgenvars
+program define _eventgenvars, rclass
+
+	#d;
+	syntax [anything] [if] [in],
+	panelvar(varname) /* Panel variable */	
+	timevar(varname) /* Time variable */
+	policyvar(varname) /* Policy variable */
+	[
+	lwindow(real 0)
+	rwindow(integer 0) /* Estimation window. Need to set a default, but it has to be based on the dataset */
+	/* since lwindow and rwindow are now optional, a default value must be provided*/
+	norm(numlist) /* Coefficients to normalize */
+	trend(string) /* Lower limit for trend */
+	
+	impute(string) /*impute policyvar. There are three options: */
+	/*nuchange  imputes outer missing values of policyvar without verifying staggered adoption*/
+	/*stag  imputes outer missing values of policyvar verifying staggered adoption*/
+	/*instag  imputes outer and inner missing values verifying staggered adoption*/
+	STatic /* Estimate static model */
+	
+
+	
+	]	
+	;
+	#d cr	
+	
+	tempvar mz mz2 kg
+	
+	marksample touse
+	
+	* mz maximum of policy outside window
+	* kg grouped event time variable, grouping dummies outside window
+	

```

Comment:
@Constantino-Carreto-Romero add an explanation for `mz2` in these comments.

Reply:
`mz2` was a variable used in the process of creation of the event-time dummies in the original code, but it is not used anymore. I will drop it from that tempvar list.

Reply:
@Constantino-Carreto-Romero  agreed, will be updated in the next commit to this PR.